server/planning/events/service.py

```diff
@@ -0,0 +1,70 @@
+from typing import AsyncGenerator, Any
+from datetime import datetime
+from superdesk.core.utils import date_to_str
+
+from planning.types import EventResourceModel
+from planning.common import get_max_recurrent_events, WORKFLOW_STATE
+from planning.core.service import BasePlanningAsyncService
+
+
+class EventsAsyncService(BasePlanningAsyncService[EventResourceModel]):
+    resource_name = "events"
+
+    async def get_expired_items(
+        self, expiry_datetime: datetime, spiked_events_only: bool = False
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        """
+        Retrieve "expired" events which are those whose end date is on or before `expiry_datetime` and
+        are not already marked as expired.
+
+        By default, items returned are:
+        - Not expired.
+        - Have an end date `<= expiry_datetime`.
+
+        If `spiked_events_only` is True, only spiked events are returned, still filtered by
+        end date `<= expiry_datetime`.
+
+        Results are sorted by start date and fetched in batches.
+        """
+        query: dict[str, Any] = {
+            "query": {
+                "bool": {
+                    "must_not": [{"term": {"expired": True}}],
+                    "filter": {"range": {"dates.end": {"lte": date_to_str(expiry_datetime)}}},
+                },
+            },
+            "sort": [{"dates.start": "asc"}],
+            "size": get_max_recurrent_events(),
+        }
+
+        if spiked_events_only:
+            del query["query"]["bool"]["must_not"]
+            query["query"]["bool"]["must"] = [{"term": {"state": WORKFLOW_STATE.SPIKED}}]
```

Comment:
in here I would suggest we use the new Enum class from `types.enums.WorkflowState` as it is being used by the model too:
```suggestion
            query["query"]["bool"]["must"] = [{"term": {"state": WorkflowState.SPIKED}}]
```